### DISCUSSION TITLE
CGI scripting

### DISCUSSION THREAD

**Discussion by dukester** [janet-lang/janet#842] [Q&A]
Is there a module for CGI scripting? Thx ...

**Comment by sogaiu** [comment]
I haven't seen one, but perhaps someone else has.

**Comment by dukester** [comment]
On Mon, 18 Oct 2021 05:46:49 -0700
sogaiu ***@***.***> wrote:

> I haven't seen one, but perhaps someone else has.

OK .. is this the best place to ask? Doesn't seem to be much
traffic here. I listen to IRC #janet but nobody seems to be there
either.
--

**Reply by uvtc** [reply]
I think it's good to ask here. I know I get notifications when there's a post here, and I'm guessing many others do as well. And, personally, I prefer the asynchronous nature of the forum.

Sorry, I don't know the answer to your question though. Maybe the docs to [Joy](https://joy.swlkr.com/) may shed some light?

**Comment by sogaiu** [comment]
@dukester If you'd like more real-time communication, there are some gitter channels (also reachable via matrix) listed here: https://janet-lang.org/#Community

**Comment by dukester** [comment]
On Mon, 18 Oct 2021 07:10:59 -0700
sogaiu ***@***.***> wrote:

> @dukester If you'd like more real-time communication, there are
> some gitter channels (also reachable via matrix) listed here:
> https://janet-lang.org/#Community

@sogaiu Don't do gitter/matrix. But thx.
--

**Comment by dukester** [comment]
On Tue, 19 Oct 2021 02:31:46 +0000 (UTC)
John Gabriele ***@***.***> wrote:

> I think it's good to ask here. I know I get notifications when
> there's a post here, and I'm guessing many others do as well.
> And, personally, I prefer the asynchronous nature of the forum.

Yes!

> Sorry, I don't know the answer to your question though. Maybe the
> docs to [Joy](https://joy.swlkr.com/) may shed some light?

Thanks for the link John!

**Reply by harryvederci** [reply]
Not really a module, but I was messing around with getting a working Janet CGI scripting setup running on a Debian server just now:

https://git.sr.ht/~harryvederci/janet-cgi-example

It's mainly for configuring the server, but the `main.janet` file does result in a working CGI response.

Notes:
- This is the first time I'm doing anything with CGI, so don't use it as a reference of how things should be done.
- There's a "TODO" item in the `00_configure-server-example-debian.bash` script about checking if I'm using a correct version of apache2.
- I did a few modifications after setting up my own configuration. It should still work, but I thought it would be polite to mention it :)

**Comment by harryvederci** [comment]
@dukester I haven't used it yet, but apparently there's a Go application with which you can serve Janet applications using CGI/SCGI/FastCGI:

https://github.com/rushsteve1/spinnerette
